### PR TITLE
Spin up WP sites with the plugin on Travis

### PR DESCRIPTION
**Pull request by nylen**
This PR adds automated WP site creation for multiple PHP and WP versions, using [lando](https://github.com/lando/lando), a tool that wraps `docker` and `docker-compose`.

This works on Travis CI and locally, as long as you [install docker-ce](https://docs.docker.com/install/linux/docker-ce/debian/#set-up-the-repository) and [install lando](https://docs.devwithlando.io/installation/installing.html).

PHP 5.2 is not supported without some extra work:  https://github.com/lando/lando/issues/1256

This is the first step towards a decent automated test matrix for this plugin.  The next steps could be something like this:

- On WP 4.9.x, install various versions of the Gutenberg plugin
- Install a tool like [puppeteer](https://github.com/GoogleChrome/puppeteer) and use it to run through some testing scenarios that verify editor controls with various settings of the plugin
- These tests could use raw Node.js scripts (probably a script that runs the major test cases in order for each WP/PHP version), or a testing framework like [mocha](https://github.com/mochajs/mocha).
- Make Travis cache the Docker images involved in this build.  More info: https://github.com/lando/lando/issues/1248

### CODE REVIEW

**Comment by pento** [comment]
Thank you for the PR, @nylen.

Lando is cool, it seems to take a lot of the hassle out of setting up a Docker environment.

Unfortunately, this PR needs a fair bit of work before it actually gives us information. As you mentioned, Puppeteer is a key part of that, confirming that the Classic Editor plugin behaves as expected.

This also runs in a fairly reactionary way: it has to be updated for every WordPress release. It'd be good to have it run against the WordPress nightly build, we can setup a daily Travis job to test that.

**Comment by nylen** [comment]
Sure, what's here so far is just the initial steps.  I'll continue working on the Puppeteer bits as I have time, unless someone wants to beat me to it of course.

6812996 adds WP nightly to the test matrix.  The next step there is setting up https://docs.travis-ci.com/user/cron-jobs/ to run a build every day.

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ env:
   - WP_VERSION=4.9.8 WP_LOCALE=es_ES PHP_VERSION=5.3
   - WP_VERSION=5.0-beta1 PHP_VERSION=5.6
   - WP_VERSION=5.0-beta2 PHP_VERSION=7.2
+  - WP_VERSION=nightly PHP_VERSION=7.2
```
[the remark this answers is not in the document]
Comment (pento) [reply .travis.yml]: https://wordpress.org/nightly-builds/wordpress-5.0-latest.zip

[the remark this answers is not in the document]
Comment (nylen) [reply .travis.yml]: Resolved in 8896d49.

**Comment by ironprogrammer** [comment]
I'm closing out this PR for now, largely due to lack of traction, and possible reconsideration of if/how this automation should be accomplished, as much time has passed since its creation.

If a renewed interest in pursuing these tests comes up, please feel free to re-open.

**Pull request closed without merging**